Activate all-access APKs

nirvn opened the pull request:
@3nids , as discussed, you'll get test APKs from this.

qfield-fairy commented:
🎉 Ta-daaa, freshly created APKs are available for 15f6397473063473d7a82049dbed213164353ed8: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3519-2b6c60f8ab5bcbb90bc81b0932c5ff21bca1aac3-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3519-2b6c60f8ab5bcbb90bc81b0932c5ff21bca1aac3-arm-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3519-2b6c60f8ab5bcbb90bc81b0932c5ff21bca1aac3-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3519-2b6c60f8ab5bcbb90bc81b0932c5ff21bca1aac3-x86-android.apk)
</details>
<!-- Sticky Pull Request Comment -->

nirvn commented:
@3nids , arm 64 all access link: http://qfieldapks.sos-ch-dk-2.exo.io/ci-builds/qfield-plus-dev-3519-2b6c60f8ab5bcbb90bc81b0932c5ff21bca1aac3-arm64-android.apk

3nids commented on .github/workflows/android.yml:
```diff
@@ -25,6 +25,14 @@ jobs:
+            all_files_access: 'ON'
+          - triplet: 'x86-android'
+            all_files_access: 'ON'
 
```
you can simplify a bit the matrix:
```suggestion
    strategy:
      matrix:
        triplet: ['arm64-android', 'arm-android', 'x64-android', 'x86-android']
        all_files_access: ['ON', 'OFF']
```

nirvn commented:
@3nids , alright, updated the workflow to add the all access APKs in the auto-generated comment. Let's see if it works.

I've also changed the suffix from "plus" to "all-access". I felt "plus" was giving a completely wrong sense of that version being somehow _better or coming with extra functionality_ when it fact it's merely about storage access.

3nids commented on .github/workflows/android.yml:
```diff
@@ -155,11 +148,11 @@ jobs:
         if: ${{ matrix.all_files_access }} == "ON"
         with:
-          name: android-plus-${{ matrix.triplet }}
+          name: android-all-access-${{ matrix.triplet }}
```
```suggestion
          name: ${{ ENV.APP_PACKAGE_NAME }}-android-${{ matrix.triplet }}
```

you can use this, and remove the other upload artifact step.

3nids commented on scripts/ci/upload_artifacts.sh:
```diff
@@ -10,16 +10,16 @@ then
-      #FILENAME_AAB="qfield-${CI_TAG}-${TRIPLET}.aab"
-      FILENAME_APK="qfield-${CI_TAG}-${TRIPLET}.apk"
+      #FILENAME_AAB="qfield-all-access-${CI_TAG}-${TRIPLET}.aab"
+      FILENAME_APK="qfield-all-access-${CI_TAG}-${TRIPLET}.apk"
```
this part is already handled in the other script, no?
you should be able to remove the conditions and just use `APP_PACKAGE_NAME`

3nids replied:
basically:

```
  if [[ "${CI_TAG}" ]];
  then
    #FILENAME_AAB="${APP_PACKAGE_NAME}-${CI_TAG}-${TRIPLET}.aab"
    FILENAME_APK="${APP_PACKAGE_NAME}-${CI_TAG}-${TRIPLET}.apk"
  else
    #FILENAME_AAB="${APP_PACKAGE_NAME}-dev-${CI_UPLOAD_ARTIFACT_ID}-${CI_COMMIT}-${TRIPLET}.aab"
    FILENAME_APK="${APP_PACKAGE_NAME}-dev-${CI_UPLOAD_ARTIFACT_ID}-${CI_COMMIT}-${TRIPLET}.apk"
  fi
```

3nids commented on .github/workflows/android.yml:
```diff
@@ -144,15 +144,7 @@ jobs:
         uses: actions/upload-artifact@v3
         if: ${{ matrix.all_files_access }} == "OFF"
```
You'll need to remove this one

3nids commented on .github/workflows/android.yml:
```diff
@@ -243,6 +228,10 @@ jobs:
             - [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/${{ env.CI_PACKAGE_FILE_BASENAME }}-arm-android.apk)
             - [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/${{ env.CI_PACKAGE_FILE_BASENAME }}-x64-android.apk)
             - [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/${{ env.CI_PACKAGE_FILE_BASENAME }}-x86-android.apk)
+            - [all access arm64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/${{ env.CI_ALL_ACCESS_PACKAGE_FILE_BASENAME }}-arm64-android.apk)
```
seems fixed now

3nids commented on scripts/ci/env_gh.sh:
```diff
@@ -44,22 +44,24 @@ fi
-  else
-    export APP_PACKAGE_NAME="qfield_dev"
-  fi
+  export APP_PACKAGE_NAME="qfield_all_access${APP_PACKAGE_NAME_SUFFIX}"
```
```suggestion
  export APP_PACKAGE_NAME="qfield${APP_PACKAGE_NAME_SUFFIX}"
```

3nids commented on scripts/ci/upload_artifacts.sh:
```diff
@@ -9,21 +9,11 @@ then
-      FILENAME_APK="qfield-dev-${CI_UPLOAD_ARTIFACT_ID}-${CI_COMMIT}-${TRIPLET}.apk"
-    fi
+    #FILENAME_AAB="${APP_PACKAGE_NAME}-dev-${CI_UPLOAD_ARTIFACT_ID}-${CI_COMMIT}-${TRIPLET}.aab"
+    FILENAME_APK="${APP_PACKAGE_NAME}-dev-${CI_UPLOAD_ARTIFACT_ID}-${CI_COMMIT}-${TRIPLET}.apk"
```
actually you can remove line 10 to 17 by this
```suggestion
    FILENAME_APK="${APP_PACKAGE_NAME}-${CI_PACKAGE_FILE_SUFFIX}-${TRIPLET}.apk"
```

3nids replied:
it's all defined in env_gh

nirvn replied:
@3nids , yeah, true.
